cli/lsp/language_server.rs

```diff
@@ -3186,6 +3187,28 @@ impl tower_lsp::LanguageServer for LanguageServer {
     }
 
     lsp_log!("Server ready.");
+
+    if matches!(
+      env::var("DENO_NO_UPDATE_CHECK"),
+      Err(env::VarError::NotPresent)
+    ) {
```

Comment:
It would be nice to keep this knowledge within `cli::tools::upgrade`. Maybe the http_client could be cloned above within the lock (so we don't have to enter it again on the next line--probably best to do that regardless), then this could be called unconditionally?

Alternatively, a `tools::upgrade::update_check_enabled()` function might be beneficial.